pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -76,6 +76,7 @@
     private BrokerEntryMetadata brokerEntryMetadata;
 
     private boolean poolMessage;
+    private boolean dlqMessage;
```

Comment:
I wonder if it is possible to not add this new field to MessageImpl.
A new field has a runtime weight.

if the problem is only about not "releasing" the message if it is send to the DLQ, what about simply "retaining" one more time before adding it to the `possibleSendToDeadLetterTopicMessages` list ?